### PR TITLE
[`flake8-use-pathlib`] Fix `PTH104`false positive when `rename` is passed a file descriptor

### LINKED ISSUE #17694: `PTH104` suggests `Path.rename()` which does not support file descriptors
### Summary

```python
import os
os.rename("src", "dst", src_dir_fd=3, dst_dir_fd=4)
```
```shell
$ ruff --version
ruff 0.11.7
$ ruff check --isolated --select PTH104 t.py
t.py:2:1: PTH104 `os.rename()` should be replaced by `Path.rename()`
  |
1 | import os
2 | os.rename("src", "dst", src_dir_fd=3, dst_dir_fd=4)
  | ^^^^^^^^^ PTH104
  |

Found 1 error.
```

`Path.rename()` has no support for `dir_fd` arguments.

Related: #17693, #12871, #7620

### Version

0.11.7

### PR DESCRIPTION
**Pull request by LaBatata101**
## Summary
Contains the same changes to the semantic type inference as https://github.com/astral-sh/ruff/pull/17705.

Fixes #17694

## Test Plan

Snapshot tests.

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs [hunk 1]
```diff
@@ -31,7 +31,35 @@ pub(crate) fn replaceable_by_pathlib(checker: &Checker, call: &ExprCall) {
...
+                // are set to non-default values.
+                // Signature as of Python 3.13 (https://docs.python.org/3/library/os.html#os.rename)
+                // ```text
+                //           0    1       2                3
+                // os.rename(src, dst, *, src_dir_fd=None, dst_dir_fd=None)
+                // ```
+                if call
+                    .arguments
+                    .find_argument_value("src_dir_fd", 2)
+                    .is_some_and(|expr| !expr.is_none_literal_expr())
```
Comment (MichaReiser) [inline crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs:45]: Can we add a test where the `src_dir_fd` is specified by position (and not name) and another test where `src_dir_fd` is `None`. The same for `dst_dir_fd`

**Review by MichaReiser — APPROVED** [review]
Thank you. Lets remove the bytes string handling for now (sorry for that)

File: crates/ruff_linter/src/rules/flake8_use_pathlib/snapshots/ruff_linter__rules__flake8_use_pathlib__tests__full_name.py.snap [hunk 2]
```diff
@@ -317,3 +317,33 @@ full_name.py:47:1: PTH123 `open()` should be replaced by `Path.open()`
    | ^^^^ PTH123
 48 | open(p, 'r', - 1, None, None, None, False, opener)
    |
+
+full_name.py:65:1: PTH123 `open()` should be replaced by `Path.open()`
```
Comment (dhruvmanila) [inline crates/ruff_linter/src/rules/flake8_use_pathlib/snapshots/ruff_linter__rules__flake8_use_pathlib__tests__full_name.py.snap:321]: I haven't looked at the issue more closely, but does the handling of bytes string is planned as a follow-up or it's to be avoided for now. Given that the tests are not removed, I'm guessing the former. If so, it'd be useful to add a TODO (or possibly remove the tests.)

**Review by dhruvmanila — APPROVED** [review]

File: crates/ruff_linter/src/rules/flake8_use_pathlib/snapshots/ruff_linter__rules__flake8_use_pathlib__tests__full_name.py.snap (hunk 2 above)
Comment (MichaReiser) [reply crates/ruff_linter/src/rules/flake8_use_pathlib/snapshots/ruff_linter__rules__flake8_use_pathlib__tests__full_name.py.snap:321]: We want to keep flagging them. So an exmaple seems fine.
